tools/clang/lib/SPIRV/DeclResultIdMapper.h

```diff
@@ -810,6 +823,25 @@ class DeclResultIdMapper {
                                          const hlsl::SigPoint *sigPoint,
                                          QualType type);
 
+  // Creates and returns a variable that is the BaseVertex builtin input. The
+  // variable is also added to the list of stage variable `this->stageVars`. Its
+  // type will be a 32-bit integer.
+  //
+  // The semantic is a lie. We currently give it the semantic for the
+  // VertexID. I'm not sure what would happen if we did not use a semantic, or
+  // tried to generate the correct one. I'm guessing there would be some issue
+  // with reflection.
+  //
+  // semantic: the semantic to attach to this variable
+  //
```

Comment:
I don't think you need to do that. I'd remove the semantic parameter, and use `{}` as location in `getBaseVertexVariable`.
I know this comment was just pulled from the function above, but seems like it was added for no real reasons except a 'what if'.
Unless somebody has an example showing this is required, I'd suggest to:
 - removing this comment on both functions
 - removing the `semantic` parameter on both functions
 - replace `*semantic` with `{}`
 - replace `semantic->loc` with `{}`
 
 (Removing this doesn't break tests, nor makes reflection fail on what I've tried so I'd assume it's safe)